mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulHome.kt

```diff
@@ -20,8 +20,11 @@ private const val SocialRoute = "social"
 /** The route associated to the settings tab. */
 private const val SettingsRoute = "settings"
 
-/** The route associated to the game tab. */
-private const val GameRoute = "game"
+/** The route associated to the play tab. */
+private const val PlayRoute = "play"
+
+/** The route associated to new game button */
+const val NewGameRoute = "new_game"
```

Comment:
I think this should be `private`. If you want to provide an action to navigate to a certain screen, you should use a callback in the signature of the stateful screen composable, so the `NavController` never leaves this `StatefulHome` composable actually.

Reply:
So we mean there is no need to pass the `NavHostController` to the `StatefulPlayScreen `which then uses it to create  a callback for the `PlayScreen`

Reply:
I don't see why would it be a bad thing to restrict the usage of the `NavController` in `StatefulHome` only. It could also be useful to pass it to `StatefulPlayScreen` for example.

Reply:
It's really a matter of encapsulating how navigation works. The fact that we're using compose's `NavController` is really an implementation detail of the `StatefulHome`; some alternatives exist, such as Accompanist's https://github.com/google/accompanist/tree/main/navigation-animation for instance, and it'd be better if we only needed to change one Composable rather than all the `StatefulXXX` composables if we migrate to it for instance (especially if we want to have some proper screen transitions at some point).

Reply:
Agree !